src/Core/Checkout/Cart/SalesChannel/CartItemAddRoute.php

```diff
@@ -92,7 +93,7 @@ public function add(Request $request, Cart $cart, SalesChannelContext $context,
 
         $cart = $this->cartCalculator->calculate($cart, $context);
         $this->cartPersister->save($cart, $context);
```

Comment:
why don't you dispatch the event in the cartPersister? so you only have one place where this event is dispatched